src/plugins/mocap/include/plugins/mocap/mocap.h

```diff
@@ -0,0 +1,406 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <array>
+#include <limits>
+
+#include "plugin_base.h"
+
+namespace mavsdk {
+
+class MocapImpl;
+class System;
+
+/**
+ * @brief This class allows users to get vehicle mocap and state information
+ * (e.g. battery, GPS, RC connection, flight mode etc.) and set mocap update rates.
+ */
+class Mocap : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto mocap = std::make_shared<Mocap>(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit Mocap(System& system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~Mocap();
+
+    /**
+     * @brief Quaternion type.
+     *
+     * All rotations and axis systems follow the right-hand rule.
+     * The Hamilton quaternion product definition is used.
+     * A zero-rotation quaternion is represented by (1,0,0,0).
+     * The quaternion could also be written as w + xi + yj + zk.
+     *
+     * For more info see: https://en.wikipedia.org/wiki/Quaternion
+     */
+    union Quaternion {
```

Comment:
Ooh, won't this cause all sort of headache for the auto-generation?

We tried to keep these data structs just plain simple without setters, etc. I think it makes it all easier.

@JonasVautherin or am I wrong?

Reply:
In this case my goal was in providing convenience for use in some math algorithms.
But if this is excessive complexity I can revert this changes of course

Reply:
Oh right, I had seen this and forgot to mention it. Yes, if that's not too annoying I'd rather having simple data structs. When they get auto-generated, we can start looking into how we could improve that.

Reply:
Reverted